test/gtest/ucm/malloc_hook.cc

```diff
@@ -799,3 +800,77 @@ UCS_TEST_F(malloc_hook, bistro_patch) {
     EXPECT_NE(patched, origin);
 #endif
 }
+
+/* test for mmap events are fired from non-direct load modules
+ * we are trying to load lib1, from lib1 load lib2, and
+ * fire mmap event from lib2 */
+UCS_TEST_F(malloc_hook, dlopen) {
+#ifndef GTEST_UCM_HOOK_LIB_DIR
+    UCS_TEST_SKIP_R("missing build configuration");
+#else
+    typedef void (fire_mmap_f)(void);
+    typedef void* (load_lib_f)(const char *path);
+
+    const char *libdlopen_test  = "/libdlopen_test.so";
+    const char *libdlopen_test2 = "/libdlopen_test2.so";
+    const char *load_lib        = "load_lib";
+    const char *fire_mmap       = "fire_mmap";
+
+    char *lib_name;
+    char *lib2_name;
+    void *lib;
+    void *lib2;
+    load_lib_f *load;
+    fire_mmap_f *fire;
+    ucs_status_t status;
+
+    status = ucm_set_event_handler(UCM_EVENT_VM_MAPPED,
+                                   0, mem_event_callback,
+                                   reinterpret_cast<void*>(this));
+    ASSERT_UCS_OK(status);
+
+    lib_name  = (char*)malloc(strlen(GTEST_UCM_HOOK_LIB_DIR) + strlen(libdlopen_test) + 1);
+    lib2_name = (char*)malloc(strlen(GTEST_UCM_HOOK_LIB_DIR) + strlen(libdlopen_test2) + 1);
+    strcat(strcpy(lib_name, GTEST_UCM_HOOK_LIB_DIR), libdlopen_test);
```

Comment:
we have std::string to to string manipulations (operator+), no need to work so hard with strcat and malloc